com.unity.render-pipelines.core/Runtime/XR/XRSystem.cs

```diff
@@ -0,0 +1,372 @@
+        static MSAASamples s_MSAASamples = MSAASamples.None;
+
+        // Internal resources used by XR rendering
+        static Material s_OcclusionMeshMaterial;
+        static Material s_MirrorViewMaterial;
+
+        // Ability to override the default XR layout
+        static Action<XRLayout, Camera> s_LayoutOverride = null;
+
+        /// <summary>
+        /// Returns true if a XR device is connected and running.
+        /// </summary>
+        static public bool displayActive
+        {
+        #if ENABLE_VR && ENABLE_XR_MODULE
+            get => (s_Display != null) ? s_Display.running : false;
+        #else
+            get => false;
+        #endif
+        }
+
+        /// <summary>
+        /// If true, the system will try to create a layout compatible with single-pass rendering.
+        /// </summary>
+        static public bool singlePassAllowed { get; set; } = true;
+
+        /// <summary>
+        /// If true, the system will log some information about the layout to the console.
+        /// </summary>
+        static public bool dumpDebugInfo { get; set; } = false;
+
+        /// <summary>
+        /// Use this method to assign the shaders that will be used to render occlusion mesh for each XRPass and the final mirror view.
+        /// </summary>
+        /// <param name="occlusionMeshPS"></param>
+        /// <param name="mirrorViewPS"></param>
+        public static void Initialize(Shader occlusionMeshPS, Shader mirrorViewPS)
+        {
+            RefreshDeviceInfo();
+
+            if (occlusionMeshPS != null)
+                s_OcclusionMeshMaterial = CoreUtils.CreateEngineMaterial(occlusionMeshPS);
+
+            if (mirrorViewPS != null)
+                s_MirrorViewMaterial = CoreUtils.CreateEngineMaterial(mirrorViewPS);
+
+            if (XRGraphicsAutomatedTests.enabled)
+                SetLayoutOverride(XRGraphicsAutomatedTests.OverrideLayout);
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to communicate to the XR device how many samples are used by MSAA.
+        /// </summary>
+        /// <param name="msaaSampleCount"></param>
+        public static void SetDisplayMSAASamples(MSAASamples msaaSamples)
+        {
+            if (s_MSAASamples == msaaSamples)
+                return;
+
+            s_MSAASamples = msaaSamples;
+
+        #if ENABLE_VR && ENABLE_XR_MODULE
+            SubsystemManager.GetInstances(s_DisplayList);
+
+            foreach (var display in s_DisplayList)
+                display.SetMSAALevel((int)s_MSAASamples);
+        #endif
+        }
+
+        /// <summary>
+        /// Returns the number of samples (MSAA) currently configured on the XR device.
+        /// </summary>
+        /// <returns></returns>
+        public static MSAASamples GetDisplayMSAASamples()
+        {
+            return s_MSAASamples;
+        }
+
+        /// <summary>
+        /// Used by the render pipeline to scale the render target on the XR device.
```

Comment:
Maybe doc could mention how to use it. I assume 1.0f corresponds to 1:1 / 100% scale but clarification could help our users

Reply:
Will change 👍